### ISSUE TITLE
PDOSWorkChain - align to fermi energy

### ISSUE THREAD

**Issue by t-reents** [aiidateam/aiida-quantumespresso#716]
In the `PDOSWorkChain` there is the argument `align_to_fermi` to align the specified energy range of the DOS/PDOS around the fermi-energy (if `True`).

https://github.com/aiidateam/aiida-quantumespresso/blob/c49def7a238617439feb75afd0d0cf9fbb681613/aiida_quantumespresso/workflows/pdos.py#L220-L230

When I look at my results this shift is not performed. Moreover, I am confused, when I look at the code, how this would work in general (maybe I am overlooking something).

https://github.com/aiidateam/aiida-quantumespresso/blob/c49def7a238617439feb75afd0d0cf9fbb681613/aiida_quantumespresso/workflows/pdos.py#L450-L476

As shown above, it is checked whether `align_to_fermi` is specified in the parameters-dict. I was wondering whether the user has to add `align_to_fermi` to the parameters-dict in the inputs for the DOS/PDOS, when creating the builder? If not, why does the code not check if `self.inputs.align_to_fermi` is True or not at this point?
So basically I am wondering when the input `align_to_fermi` is actually used?

**Comment by sphuber** [comment]
Hi @t-reents, thanks for the report. There indeed seems to be a problem here as the `align_to_fermi` input is completely ignored. With the current code, you can simply get it to run by adding it to the `dos.parameters` and `pdos.parameters` namespaces, e.g.:
````python
inputs = {
    'dos': {
        'parameters': Dict(dict={
            ...
            'align_to_fermi': True
        }
    }
}
```
That being said, I am not sure what the actual solution should be. It seems the top-level `align_to_fermi` input can simply be removed. I think it merely complicates the interface because since it potentially applies to both the `dos` and `pdos` calculation, and the value can also be specified for each explicit step, defining which dominates in all possible combinations becomes confusing. Also the help string of the top-level input is incomplete. It seems to be referencing the validation of the `align_to_fermi` key, but _only_ the one specified in the respective `parameters` dict, not the top-level one. @chrisjsewell what was your original reasoning here?

**Comment by t-reents** [comment]
Hi @sphuber thanks for your quick reply. Sure this way it will run, was just a little bit confused whether I am overlooking something or if it's indeed ignored.

Thank you!

**Comment by sphuber** [comment]
Yeah, that seems like a mistake and your correction is right.

**Comment by mbercx** [comment]
Thanks for the report @t-reents! This does seem to be the result of a bit of rushed refactoring when we were planning to quickly use this work chain for a project. I think the purpose of the `align_to_fermi` as a top-level input is to allow the user to specify that the `Emin` and `Emax` inputs defined in the respective parameters of the `DosCalculation` and `ProjwfcCalculation` should be considered as relative to the Fermi level obtained from the NSCF. However, the conversion from having these _all_ (i.e. `align_to_fermi`, `Emin` and `Emax`) as part of a top-level input shared by the `dos.x` and `projwfc.x` calculations to exposing the inputs of these calculations and only having `align_to_fermi` as a top-level input clearly messed things up.

I'm not sure if the `align_to_fermi` should be provided in the parameters of the `ProjwfcCalculation` and `DosCalculation`. These are the exposed parameters of these respective calculations and so I don't think it has a place there (the UX also seems bad, how would the user know that this is even possible?). 

However, the mixing of the top-level `align_to_fermi` and then extracting the `Emin` and `Emax` from the actual parameters also doesn't seem very intuitive. So maybe for this use case it would make sense to have a top-level input where the user can specify the desired energy range around the Fermi level (e.g. a `fermi_energy_range` `List` node?). If this input is provided, the work chain will set `Emin` and `Emax` for both post-processing calculations based on the Fermi level from the NSCF and this energy range.

To summarize, the usage should be:

1. The default `Emin` and `Emax` should be extracted from the NSCF.
2. If the user _does_ provide `Emin` and `Emax` in the parameters, these should override the defaults.
3. If the user provides the top-level `fermi_energy_range` input, `Emin` and `Emax` will be determined using the Fermi level from the NSCF and the energy range specified.

We can then also add validation that not both `fermi_energy_range` and `Emin`/`Emax` in the parameters are specified. In this case the user-provided `Emin`/`Emax` values will be overridden and the user should be warned of this.

**Comment by t-reents** [comment]
Fixed in #764
